src/main/kotlin/app/Configurator.kt

```diff
@@ -0,0 +1,178 @@
+    // order: current, local, user, default.
+    private fun mergeLevels(): Options {
+        val levels = arrayListOf(current, config, default)
+        val merged = Options()
+
+        for (level in levels) {
+            if (merged.username == null) {
+                merged.username = level.username
+            }
+            if (merged.password == null) {
+                merged.password = level.password
+            }
+            if (merged.silent == null) {
+                merged.silent = level.silent
+            }
+        }
+
+        return merged
+    }
+
+    // Loads config file from specified path.
+    private fun loadConfig(path: String): Options {
+        val options = Options()
+
+        try {
+            val file = File(path, configFileName)
+
+            if (!file.exists() || !file.isFile) {
+                return options  // No configuration file
+            }
+
+            val config = PropertiesConfiguration(file)
+
+            // Accessing configuration properties. Unknown values should be
+            // null to distinguish them from specified values from other levels.
+            options.username = config.getString(configUsername, null)
+            options.password = config.getString(configPassword, null)
+            options.silent = config.getBoolean(configSilent, null)
+        }
+        catch (e: ConfigurationException) {
+            // Error while loading the properties file.
+        }
+        catch (e: SecurityException) {
+            // Read access denied.
+        }
+
+        return options
+    }
+
+    // Saves config file to specified path. Returns true on success.
+    private fun saveConfig(path: String, options: Options): Boolean {
+        try {
+            val file = File(path, configFileName)
+
+            file.createNewFile()  // Creates a new file if it didn't exist.
+
+            val config = PropertiesConfiguration(file)
+
+            if (options.username != null) {
+                config.setProperty(configUsername, options.username)
+            }
+            if (options.password != null) {
+                config.setProperty(configPassword, options.password)
+            }
+            if (options.silent != null) {
+                config.setProperty(configSilent, options.silent)
+            }
+
+            config.save(file)
+
+            return true
+        }
+        catch (e: IOException) {
+            // IO error occurred.
+        }
+        catch (e: SecurityException) {
+            // Read access denied.
+        }
+        catch (e: ConfigurationException) {
+            // Error while loading or saving the properties file.
+        }
+
+        return false
+    }
+
+    fun setCurrentOptions(options: Options) {
+        current = options
+    }
+
+    fun loadUserLevelConfig() {
+        if (userDir != null) {
+            config = loadConfig(userDir)
+        }
+    }
+
+    fun saveUserLevelConfig(options: Options): Boolean {
```

Comment:
I'll rename the function, but we need it still because it's being called from Main when config command is used.